modules/identity/src/Volo.Abp.Identity.AspNetCore/Volo/Abp/Identity/AspNetCore/AbpSignInManager.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Volo.Abp.Identity.AspNetCore
+{
+    public class AbpSignInManager : SignInManager<IdentityUser>
+    {
+        protected AbpIdentityAspNetCoreOptions AbpOptions { get; }
+
+        public AbpSignInManager(
+            IdentityUserManager userManager,
+            IHttpContextAccessor contextAccessor,
+            IUserClaimsPrincipalFactory<IdentityUser> claimsFactory,
+            IOptions<IdentityOptions> optionsAccessor,
+            ILogger<SignInManager<IdentityUser>> logger,
+            IAuthenticationSchemeProvider schemes,
+            IUserConfirmation<IdentityUser> confirmation,
+            IOptions<AbpIdentityAspNetCoreOptions> options
+        ) : base(
+            userManager,
+            contextAccessor,
+            claimsFactory,
+            optionsAccessor,
+            logger,
+            schemes,
+            confirmation)
+        {
+            AbpOptions = options.Value;
+        }
+
+        public override async Task<SignInResult> PasswordSignInAsync(
+            string userName,
+            string password,
+            bool isPersistent,
+            bool lockoutOnFailure)
+        {
+            foreach (var externalLoginProviderInfo in AbpOptions.ExternalLoginProviders.Values)
+            {
+                var externalLoginProvider = (IExternalLoginProvider) Context.RequestServices
+                    .GetRequiredService(externalLoginProviderInfo.Type);
+
+                if (await externalLoginProvider.TryAuthenticateAsync(userName, password))
+                {
+                    var user = await UserManager.FindByNameAsync(userName);
```

Comment:
There may be a user with the same name already in the application.

Reply:
Should we only query user that `IsExternal` is `true`?

Reply:
I don't think it is necessary. If you configured LDAP as an external login provider later and username & pass matches to a user in the LDAP, then we can convert this to external user (see https://github.com/abpframework/abp/pull/4979/files/2ad0beb94a29c5876b0003da3021a56e023d2e74#diff-b4b68ccdc1d9c7beca4d9c3e3f329ba3R133).